common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseController.java

```diff
@@ -186,6 +186,11 @@ protected TokenResult performTokenRequest(@NonNull final OAuth2Strategy strategy
         HttpWebRequest.throwIfNetworkNotAvailable(parameters.getAppContext());
 
         TokenRequest tokenRequest = strategy.createTokenRequest(request, response);
+
+        if (tokenRequest instanceof MicrosoftTokenRequest) {
```

Comment:
should we do this in getAuthorizationRequest? (adding claims as part of the AuthorizationRequest object, then use those values to construct TokenRequest in strategy.createTokenRequest()).

Lengthier solution, but I think it fits with the pattern we have here better.

Reply:
Actually... I just noticed that we already set the claims over there.

`

            // Add additional fields to the AuthorizationRequest.Builder to support interactive
            builder.setLoginHint(
                    acquireTokenOperationParameters.getLoginHint()
            ).setExtraQueryParams(
                    acquireTokenOperationParameters.getExtraQueryStringParameters()
            ).setPrompt(
                    acquireTokenOperationParameters.getOpenIdConnectPromptParameter().toString()
            ).setClaims(
                    parameters.getClaimsRequestJson()
            ).setRequestHeaders(
                    acquireTokenOperationParameters.getRequestHeaders()
            );

`

Reply:
Addressed